src/models/fields/base-field-configuration.ts

```diff
@@ -1,36 +1,96 @@
+      is: 'number',
+      then: Joi.number().default(0),
+    })
+    .when(Joi.ref('type'), {
+      is: 'integer',
+      then: Joi.number().integer().default(0),
+    })
+    .when(Joi.ref('type'), {
+      is: 'boolean',
+      then: Joi.boolean().default(0),
+    })
+    .required(),
   valueOptions: Joi.array().items(Joi.object().keys({
-    text: Joi.string().required(),
+    text: stringOrTranslationsSchema.required(),
     value: baseFieldSchema.required().description('Will be passed through parser'),
-  })).allow(null)
-    .description('If null, inputType should not be select or radio. If not null, input type should be select or radio'),
-  prefix: Joi.string().description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
-  suffix: Joi.string().description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
+  })).allow(null).default(null)
+    .description('If null, inputType should not be select or radio. If not null, input type should be select or radio. Default: null'),
+  regex: Joi.string().description('If provided, type should be string and the provided value should adhere to this regex'),
+  lowerbound: Joi.number().description('If provided, type should be number or integer (and lowerbound should be integer as well in case type is integer) and provided value should not be lower than this value'),
+  upperbound: Joi.number().description('If provided, type should be number or integer (and lowerbound should be integer as well in case type is integer) and provided value should not be higher than this value'),
+  showIf: Joi.object().keys({
+    key: Joi.string().required(),
+    value: Joi.alternatives(
+      Joi.number(),
+      Joi.string(),
+      Joi.boolean(),
+    ).required(),
+  }).description('Show this field if other field with provided key is set to provided value. Referenced field must exists already'),
+  prefix: stringOrTranslationsSchema.description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
+  suffix: stringOrTranslationsSchema.description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
   hint: Joi.string().allow('').description('As shown near the input field'),
 })
   .tag('baseFieldConfiguration')
   .description('Defines which data can be stored in form fields.');
 
 interface ValueOption {
-  text: string;
+  // string for backwards compatibility, shown at all locales
+  text: string | Translations;
   value: BaseField;
 }
 
 interface BaseFieldConfiguration {
   key: string;
-  name: string;
-  inputType?: 'text' | 'textarea' | 'select' | 'radio' | 'switch' | 'checkbox' | 'file' | 'files';
+  type: 'string' | 'boolean' | 'number' | 'integer';
+  name: string | Translations;
+  inputType: 'text' | 'textarea' | 'select' | 'radio' | 'switch' | 'checkbox' | 'file' | 'files';
+  defaultValue?: string | boolean | number;
```

Comment:
The problem with default values, is that for the client, the keys are optional, while for the backend they are always present. I propose to use the TS Interface for the backend, so with `defaultValue` and `valueOptions` being required. A client that is using TS will then not have the benefit of default values (ie. he has to supply values himself), but I think that outweighs the extra checks the backend has to do.